tests/foreman/cli/test_classparameters.py

```diff
@@ -349,60 +137,22 @@ def test_positive_list_by_hostgroup_id(self):
             'id': sc_param_id,
         })
         self.assertEqual(sc_param['override'], True)
-        hostgroup = make_hostgroup({
-            'environment-id': self.env['id'],
-            'puppet-class-ids': self.puppet_class['id']
-        })
-        hostgroup_sc_params = SmartClassParameter.list(
-            {'hostgroup-id': hostgroup['id']})
-        self.assertGreater(len(hostgroup_sc_params), 0)
-        self.assertIn(sc_param_id, [scp['id'] for scp in hostgroup_sc_params])
-        # Check that only unique results are returned
-        self.assertEqual(
-            len(hostgroup_sc_params),
-            len({scp['id'] for scp in hostgroup_sc_params})
-        )
-
-    @tier1
-    @upgrade
-    def test_positive_list_by_puppetclass_name(self):
-        """List all the parameters for specific puppet class by name.
-
-        :id: 6d62968f-dc5b-4d7f-ac21-c1335a827960
-
-        :expectedresults: Parameters listed for specific Puppet class.
-
-        :CaseImportance: Critical
-        """
-        sc_params = SmartClassParameter.list(
-            {'puppet-class': self.puppet_class['name']})
-        self.assertGreater(len(sc_params), 0)
-        self.assertTrue(
-            {scp['id'] for scp in self.sc_params_list}.issubset(
-                {scp['id'] for scp in sc_params})
-        )
-        # Check that only unique results are returned
-        self.assertEqual(len(sc_params), len({scp['id'] for scp in sc_params}))
-
-    @tier1
-    def test_positive_list_by_puppetclass_id(self):
-        """List all the parameters for specific puppet class by id.
 
-        :id: a7a8af1a-514b-4910-9e19-75306f634041
+        # check listing parameters for selected queries
+        for query in list_queries:
```

Comment:
I wonder if you shouldn't use [subtests](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) here... or shuffle list before using it...

Imagine querying by environment is broken and it takes two months to provide fix. During these two months, we will know nothing about querying by host or puppet class, which might or might not be broken. If you use subtest, we should get separate failure for each query; if you shuffle list, we at least will have a chance to try other queries before encountering broken one and failing whole test.

The problem with subtests approach is that it won't work in pytest, as pytest doesn't support unittest subtests natively. There is `pytest-subtests` plugin, but it requires pytest 4.4 - newer than what we use. Of course updating pytest is out of scope for your PR.

To sum up: conceptually, I would prefer this part to not be blocked by any single failure. What do you think?

Reply:
Good point, added the subTest. I think we can deal with the pytest problem when it arises, cheers